From 6c58fd21241fe7121758bb67dcd347c84d15267a Mon Sep 17 00:00:00 2001
From: Evgeniy Didin <Evgeniy.Didin@synopsys.com>
Date: Mon, 23 Apr 2018 20:35:33 +0300
Subject: [PATCH] Fix wrong mode select in fbdev-drm emulation

Running DOOM (uses SDL), there is incorrect select of color encoding which leads to 
incorrect display colors. SDL passes zeroed structure "var" to check function, so 
here we add hack to consider this case. 

Signed-off-by: Evgeniy Didin <didin@synopsys.com>
---
 drivers/gpu/drm/drm_fb_helper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_fb_helper.c b/drivers/gpu/drm/drm_fb_helper.c
index 0646b108030b..a25e2d5f1a24 100644
--- a/drivers/gpu/drm/drm_fb_helper.c
+++ b/drivers/gpu/drm/drm_fb_helper.c
@@ -1593,7 +1593,7 @@ int drm_fb_helper_check_var(struct fb_var_screeninfo *var,
 
 	switch (var->bits_per_pixel) {
 	case 16:
-		depth = (var->green.length == 6) ? 16 : 15;
+		depth = (var->green.length == 6 || var->green.length == 0) ? 16 : 15;
 		break;
 	case 32:
 		depth = (var->transp.length > 0) ? 32 : 24;
-- 
2.16.2

